service/src/main/java/uk/nhs/adaptors/gp2gp/transformJsonToXmlTool/TransformJsonToXml.java

```diff
@@ -0,0 +1,237 @@
+import uk.nhs.adaptors.gp2gp.ehr.mapper.ObservationToNarrativeStatementMapper;
+import uk.nhs.adaptors.gp2gp.ehr.mapper.OutputMessageWrapperMapper;
+import uk.nhs.adaptors.gp2gp.ehr.mapper.ParticipantMapper;
+import uk.nhs.adaptors.gp2gp.ehr.mapper.PertinentInformationObservationValueMapper;
+import uk.nhs.adaptors.gp2gp.ehr.mapper.RequestStatementMapper;
+import uk.nhs.adaptors.gp2gp.ehr.mapper.StructuredObservationValueMapper;
+import uk.nhs.adaptors.gp2gp.ehr.mapper.SupportedContentTypes;
+import uk.nhs.adaptors.gp2gp.ehr.mapper.diagnosticreport.DiagnosticReportMapper;
+import uk.nhs.adaptors.gp2gp.ehr.mapper.diagnosticreport.ObservationMapper;
+import uk.nhs.adaptors.gp2gp.ehr.mapper.diagnosticreport.SpecimenMapper;
+import uk.nhs.adaptors.gp2gp.ehr.mapper.parameters.EhrExtractTemplateParameters;
+import uk.nhs.adaptors.gp2gp.gpc.GetGpcStructuredTaskDefinition;
+
+import java.io.BufferedWriter;
+import java.io.File;
+import java.io.FileWriter;
+import java.io.IOException;
+import java.lang.invoke.MethodHandles;
+import java.nio.charset.StandardCharsets;
+import java.nio.file.Files;
+import java.nio.file.Paths;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+import java.util.Optional;
+
+public class TransformJsonToXml {
+
+    private static final String JSON_FILE_INPUT_PATH =
+            Paths.get("src/").toFile().getAbsoluteFile().getAbsolutePath() + "/../../transformJsonToXml/input/";
+    private static final String XML_OUTPUT_PATH =
+            Paths.get("src/").toFile().getAbsoluteFile().getAbsolutePath() + "/../../transformJsonToXml/output/";
+    private static final Logger LOGGER = LoggerFactory.getLogger(MethodHandles.lookup().lookupClass());
+    private static final FhirParseService FHIR_PARSE_SERVICE = new FhirParseService();
+
+
+    public static void main(String[] args) throws Exception {
+        String startTest = (System.getenv().getOrDefault("JSON_TO_XML_START_TOOL", "False"));
+        LOGGER.error("variable: " + startTest);
+        try {
+            if (startTest.equals("True")) {
+                var inputWrapper = getFiles();
+                for (int i = 0; i < inputWrapper.getJsonFileInputs().size(); i++) {
+                    var jsonString = inputWrapper.getJsonFileInputs().get(i);
+                    String xmlResult = mapJsonToXml(jsonString);
+                    var fileName = inputWrapper.getJsonFileNames().get(i);
+                    writeToFile(xmlResult, fileName);
+                }
+            }
+        } catch (Exception e) {
+            LOGGER.error("error: " + e.getMessage());
+        }
+        LOGGER.error("end");
+    }
+
+    private static InputWrapper getFiles() throws Exception {
+        File[] files = new File(JSON_FILE_INPUT_PATH).listFiles();
+        List<String> jsonStringInputs = new ArrayList<>();
+        List<String> fileNames = new ArrayList<>();
+
+        assert files != null;
+        LOGGER.error(JSON_FILE_INPUT_PATH);
```

Comment:
The log is displays as error, while it could be valid case as well so I would change it to info and provide something that gives more information.
I would also change the order here, to check files number first and do not use assert (as I don't think is gives us the best information what is the problem)

for example:
```
if (files  == null || files.length == 0) {
    throw new Exception("No json files found");
}
LOGGER.info("Processing " + files.length + " files from location: " + JSON_FILE_INPUT_PATH);
```